From ddca29389d87f6bfaa45e36e04d1be1ffd1948ac Mon Sep 17 00:00:00 2001
From: Luwei Zhou <b45643@freescale.com>
Date: Wed, 30 Oct 2013 14:24:34 +0800
Subject: [PATCH 0573/1074] ENGR00285441 mxc: mlb: Fix the receive issue of
 ISOC mode on Sabreauto

This patch fixes the receive issue of MLB ISO mode on Sabreauto side.
During debug, it is found that the Sabreauto can only receive
1/5 packets of MITB sends.The MITB sends out 188 bytes per packet.
The Sabreauto receives 188*5 bytes. It seems that the packet length
on Sabreauto side is 5 times larger than MITB side. After changing
the defautl block number to 1, Sabreauto passes the test.

Signed-off-by: Luwei Zhou <b45643@freescale.com>
[Original patch taken from git://git.freescale.com/imx/linux-2.6-imx.git]
Signed-off-by: Biyao Zhai <biyao.zhai@windriver.com>
---
 drivers/mxc/mlb/mxc_mlb150.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/mxc/mlb/mxc_mlb150.c b/drivers/mxc/mlb/mxc_mlb150.c
index c2f418c..79aeeb5 100755
--- a/drivers/mxc/mlb/mxc_mlb150.c
+++ b/drivers/mxc/mlb/mxc_mlb150.c
@@ -176,7 +176,7 @@
 #define CH_ISOC_BLK_SIZE_188	(188)
 #define CH_ISOC_BLK_SIZE_196	(196)
 #define CH_ISOC_BLK_SIZE	(CH_ISOC_BLK_SIZE_188)
-#define CH_ISOC_BLK_NUM		(5)
+#define CH_ISOC_BLK_NUM		(1)
 #define CH_ISOC_CDT_BUF_DEP	(CH_ISOC_BLK_SIZE * CH_ISOC_BLK_NUM)
 #define CH_ISOC_ADT_BUF_DEP	(CH_ISOC_CDT_BUF_DEP)
 #define CH_ISOC_BUF_SZ		(1024)
-- 
1.7.5.4

